docs/spec_draft.md

```diff
@@ -1266,6 +1267,56 @@ More formally, `result[i0, ..., iR-1]` is equal to:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.reduce
+
+### Semantics
+
+Applies a `computation` function to `inputs` and `init_values` along the
+`dimensions` and produces a `result` tensor.
+
+### Inputs
+
+| Name          | Type                                             |
+|---------------|--------------------------------------------------|
+| `inputs`      | variadic number of tensors of any supported type |
+| `init_values` | variadic number of tensors of any supported type |
+| `dimensions`  | 1-dimensional tensor constant of type `si64`     |
+| `computation` | `function`                                       |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
```

Comment:
Ca we say something about the shape of any result[k] ? Something along the lines of 
https://github.com/tensorflow/tensorflow/blob/ab758b2c95382eeba943c741e7f8f3d9ba1b4f2b/tensorflow/compiler/xla/service/shape_inference.cc#L2304